### PR TITLE
Feat/output yaml

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

**Comment by mzedel** [comment]
@mender-test-bot start pipeline

**Comment by mzedel** [comment]
> @mender-test-bot start pipeline

**Comment by mzedel** [comment]
> > @mender-test-bot start pipeline --pr meta-mender/2151

**Comment by mzedel** [comment]
> > > @mender-test-bot start pipeline --pr meta-mender/2151

File: cli/read.go [hunk 1]
```diff
@@ -33,6 +33,156 @@ import (
...
+	printDepends(p, indentationLevel)
+	printClearsProvides(p, indentationLevel)
+	printUpdateMetadata(p, indentationLevel)
+	printFiles(p.GetUpdateAllFiles(), indentationLevel)
+}
+
+func printUpdates(updatePayloads map[int]handlers.Installer, indentationLevel int) {
+	fmt.Printf("%sUpdates:\n", strings.Repeat(defaultIndentation, indentationLevel))
+	for k, payload := range updatePayloads {
+		fmt.Printf("%sUpdate-%d:\n", strings.Repeat(defaultIndentation, indentationLevel+1), k)
```
Comment (lluiscampos) [inline cli/read.go:212]: (relates to the previous comment).

Here I suggest either a list of maps, as suggested for `Files`, or to use strings like `0000`, `0001`, etc to align with the Artifact protocol definition.

No solution is perfect, but both are better that introducing a prefix `Update-` IMO.

File: cli/read.go [hunk 2]
```diff
@@ -94,136 +244,29 @@ func readArtifact(c *cli.Context) error {
...
+	printStateScripts(scripts, 1)
+	fmt.Println()
+	updatePayloads := ar.GetHandlers()
+	printUpdates(updatePayloads, 0)
 
-	fmt.Printf("\nUpdates:\n")
-	for k, p := range inst {
-		printPayload(k, p)
-	}
 	return nil
```
Comment (lluiscampos) [inline cli/read.go:302]: This main read function reads now so beautiful :heart_eyes:

File: cli/read.go [hunk 3]
```diff
@@ -33,6 +33,156 @@ import (
...
+	fmt.Printf("%sVersion: %d\n", strings.Repeat(defaultIndentation, indentationLevel+1), info.Version)
+	fmt.Printf("%sSignature: %s\n", strings.Repeat(defaultIndentation, indentationLevel+1), sigInfo)
+	fmt.Printf("%sCompatible devices:\n", strings.Repeat(defaultIndentation, indentationLevel+1))
+	for _, deviceType := range ar.GetCompatibleDevices() {
+		fmt.Printf("%s- %s\n", strings.Repeat(defaultIndentation, indentationLevel+2), deviceType)
+	}
+}
+
+func printStateScripts(scripts []string, indentationLevel int) {
+	if len(scripts) > -1 {
```
Comment (lluiscampos) [inline cli/read.go]: Could we use `len(scripts) == 0`, like in the rest of the checks below Or the opposite for all of them?

File: cli/modify_existing_test.go [hunk 4]
```diff
@@ -301,9 +301,9 @@ yOTl4wVLQKA6mFvMV9o8B9yTBNg3mQS0vA==
 func removeVolatileEntries(input string) string {
 	var output strings.Builder
 	for _, line := range strings.Split(input, "\n") {
-		if strings.HasPrefix(line, "      checksum:") ||
-			strings.HasPrefix(line, "      modified:") ||
-			strings.HasPrefix(line, "\trootfs-image.checksum:") {
+		if strings.Contains(line, " checksum:") ||
+			strings.Contains(line, " modified:") ||
+			strings.Contains(line, "rootfs-image.checksum:") {
```
Comment (lluiscampos) [inline cli/modify_existing_test.go:309]: This commit (fixing the tests) should be squashed with the first one (changing the output).

File: cli/modify_existing_test.go [hunk 5]
```diff
@@ -16,7 +16,6 @@ package cli
 
 import (
 	"io"
-	"io/ioutil"
```
Comment (lluiscampos) [inline cli/modify_existing_test.go:19]: :partying_face:

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -134,6 +136,7 @@ test:smoketests:linux:
     - ./mender-artifact-linux write rootfs-image -t test -o test-rfs.mender -n test -f test.txt
     - ./mender-artifact-linux read test-rfs.mender
     - ./mender-artifact-linux validate test-rfs.mender
+    - test $(./mender-artifact-linux read --no-progress test-rfs.mender | yq eval -o json | jq -r '."Mender artifact".Name') == "test"
```
Comment (lluiscampos) [inline .gitlab-ci.yml:139]: Really nice test :rocket:

File: cli/read.go [hunk 7]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 Northern.tech AS
+// Copyright 2024 Northern.tech AS
```
Comment (lluiscampos) [inline cli/read.go:1]: For future reference. We don't need to modify the copyright year anymore in the source files - we removed that check some time ago.

I remember from your WIP PR that you were having CI failures, but AFAIK this should not be the culprit.

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Thank you very much for your contribution :medal_sports: 

Requesting changes to discuss the `Files` and `Updates` list vs map oputut :memo: Otherwise is good to go!

File: cli/read.go (hunk 7 above)
Comment (mzedel) [reply cli/read.go:1]: Thanks for the confirmation - I'm happy to hear that the simplification is valid here as well! When I got the CI failures related to this I wasn't sure, but I guess I started from a stale repo state and that's the culprit 😬.

File: cli/read.go (hunk 3 above)
Comment (mzedel) [reply cli/read.go]: That should be done in e10594ce2c232d134b0ec91c7ea96a3d62c50ae7 👌

File: cli/read.go (hunk 1 above)
Comment (mzedel) [reply cli/read.go:212]: There used to be "space prefixed integers", but these would disturb the formatting, I'll change align it with your excellent `Files` suggestion!

**Comment by mzedel** [comment]
> > > > @mender-test-bot start pipeline --pr meta-mender/2151

**Review by lluiscampos — APPROVED** [review]

**Comment by lluiscampos** [comment]
@mzedel There was one CI job that timed out and I just restarted it. This I haven't seen in many months.

**Pull request merged**